### ISSUE TITLE
Fix failing tests

### ISSUE THREAD

**Issue by abdurraheemali** [abdurraheemali/historical-action-predictor#1]

**Comment by abdurraheemali** [comment]
This is the highest priority item right now. Taking a closer look, there are several issues... first of all we don't even detect the tests automatically, so it becomes a two click process instead of one. I should update some config file to make sure they are available right away.


![image](https://github.com/abdurraheemali/historical-action-predictor/assets/98995519/998254ab-fc46-4dc1-913b-e19f0bc3d2f8)

Secondly these tests don't fail consistently across runs, if you retry a couple of times then it can be 12 or 13.

![image](https://github.com/abdurraheemali/historical-action-predictor/assets/98995519/8f215f19-a0ad-4f87-80df-d1954ed232bf)

![image](https://github.com/abdurraheemali/historical-action-predictor/assets/98995519/bc02b5cc-5573-44bd-95e5-655f4fa3b100)

Third these tests take way too long, 2.5s on the last run, might want to fix that then add benchmarks to prevent a future regression.

**Comment by abdurraheemali** [comment]
Ok one problem is that the tensor sizes aren't matching, probably a bug in the implementation.

Another is that the tests themselves aren't correct. Like, over here the check for myopia is wrong.

[# Check for signs of long-term planning or distributional shifts](https://github.com/abdurraheemali/historical-action-predictor/blob/7e628fb066453b2b6b5421d5f0848592513a9e8e/training/test_action_predictor.py#L143-L146)

Which makes the green tests also suspect. We could get this to pass by setting a more generous comparison threshold, but the whole point of the experiment is to understand what the range for those would be. The tests were added too early and don't represent a clear understanding of the theory.

**Comment by abdurraheemali** [comment]
Taking a step back, is the code actually implementing the zero sum training algorithm properly?

https://www.lesswrong.com/posts/A48amesEmqD8KNSmY/conditional-prediction-with-zero-sum-training-solves-self#Comparisons_to_Other_Approaches

> Another way to view the zero-sum game, in the case where both players are the same model, is as below optimization problem (where q is assumed to be the ground truth). Note that we are here just subtracting the score received by the same model, but we are fixing that score when optimizing p to avoid making the objective 0.)

![image](https://github.com/abdurraheemali/historical-action-predictor/assets/98995519/5023362a-4d6c-46ab-a034-a2dfa15cc684)

https://github.com/abdurraheemali/historical-action-predictor/blob/7e628fb066453b2b6b5421d5f0848592513a9e8e/training/train.py#L94-L123

The gradient update is supposed to be solely based on the action taken.

What is zerosum_running_score even doing? Hmm, no, this needs to be rewritten.

**Comment by abdurraheemali** [comment]
@ShaikhTheWorld thanks for fixing the training code.

**Comment by abdurraheemali** [comment]
ah, seems like the tests aren't taking the pretrained zero sum models but instead initializing with random weights and then running for one epoch. should fix that.

**Comment by abdurraheemali** [comment]
also the "test private information" does *not* do what it says on the tin, which I would expect to be some PPML stuff. gotta work on that toooo.
